docs/advanced/customsuseimages.md

```diff
@@ -0,0 +1,34 @@
+---
+id: customsuseimages
+sidebar_position: 8
+sidebar_label: Custom SUSE VM Images
+title: "Custom SUSE VM Images"
+keywords:
+- Custom Images
+Description: How to create custom SLES and openSUSE guest virtual machine images
+---
+
+<head>
+  <link rel="canonical" href="https://docs.harvesterhci.io/v1.3/advanced/customsuseimages"/>
+</head>
+
+SUSE provides SUSE Linux Enterprise (SLE) and openSUSE Leap virtual machine (VM) images suitable for use in Harvester.  These images are built on the [openSUSE Build Service](https://build.opensuse.org/) (OBS) using the [Kiwi](https://osinside.github.io/kiwi/) image building tool.
+
+SLE and openSUSE VM images can be downloaded and used as-is from https://www.suse.com/download/sles/ and https://get.opensuse.org/leap/ respectively.  You will ordiniarly want to use the Minimal VM Cloud qcow2 images, as these include the cloud-init tool necessary for automatically configuring the VM.  Other image variants require logging into the VM console to perform initial configuration.
+
+:::info
+Prior to SLES 15 SP5 and openSUSE 15.5 the Minimal VM Cloud images were named Minimal VM OpenStack Cloud.
+:::
+
+If you wish to create custom VM images based on the above, for example to include additional packages, you can do so based on the appropriate Kiwi template.
+
+The easiest way to do this is to use the openSUSE Build Service, starting at https://build.opensuse.org/image_templates.  For example, to create an image based on SLE 15 SP5, you would select "SLE 15 SP5 Minimal" under the "Templates using SUSE Linux Enterprise 15 SP5" heading then click the "Create appliance" button.  From there, OBS provides a convenient user interface for adding additional packages.  Your image will be automatically built, and can be downloaded from OBS, then you can upload it into Harvester.  More information about OBS is available in the [User Guide](https://openbuildservice.org/help/manuals/obs-user-guide/).
```

Comment:
While this seems obvious to us, do you think it's a better idea to have a step-by-step tutorial with screenshots to guide the user/customer? The reason of my suggestion is I tested the template and I noticed some obstacles and learning path for me (as an OBS newbie 😅 )
- Where should I start to add my packages: by digging around it seems to be the [`Minimal.kiwi` file](https://build.opensuse.org/package/view_file/home:kieferchang:branches:SUSE:Templates:Images:SLE-15-SP5/kiwi-templates-Minimal/Minimal.kiwi?expand=1).
- I saw some profiles, and I figured out the current profile could be `kvm-and-xen` according to line 3.
- Then I guess can add packages in `<packages></packages>` with profiles.
- Then my question becomes: can I just add a package or do I have to get a SLES subscription first?
- I notice the package is built and an image is generated. But I have to turn on the publish flag in the upper project to enable public downloading.
- I can't boot the image in Harvester, still figuring out:
<img width="886" alt="截圖 2024-01-08 上午10 45 28" src="https://github.com/harvester/docs/assets/1691518/02b0ccc3-392d-4331-9435-a433880aa665">

Reply:
Good idea - I really wasn't sure how far to go with that, but you're right, with no OBS experience it would be very helpful to make it step by step with screenshots.

To answer one of those questions, yes the `.kiwi` file is where you can add packages, but OBS actually also has a special image view if you started with the template, that makes this a bit easier:

![image](https://github.com/harvester/docs/assets/754700/20c64358-aab6-4869-966f-5c5d0542b0f2)

That's accessible if you click "View Image" under Actions on the LHS of the screen (in the above screenshot it's "View Package" to get you back to the list of files).

That failure you're getting booting the image is going to be disk size - see https://github.com/harvester/harvester/issues/4905

Reply:
Also, nit picking: could you remove the double spaces after the period, please Tim?

Reply:
> Also, nit picking: could you remove the double spaces after the period, please Tim?

Oh not that again. Do you have any idea how many years I've had that particular bit of muscle memory? ;-)